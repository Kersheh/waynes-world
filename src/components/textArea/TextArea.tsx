import React from 'react';
import { useFormContext } from 'react-hook-form';

import styles from './TextArea.module.scss';

interface TextAreaProps {
  name: string;
  label?: string;
  required?: boolean;
  maxLength?: number;
  onChange?: (event: React.ChangeEvent<HTMLTextAreaElement>) => void;
}
const TextArea = ({
  name,
  label,
  required,
  maxLength,
  onChange
}: TextAreaProps) => {
  const { watch, register } = useFormContext();

  return (
    <div className={styles.textArea}>
      {label && (
        <div className={styles.label}>
          {label}
          {required && ' *'}
        </div>
      )}
      <textarea
        name={name}
        required={required}
        maxLength={maxLength}
        ref={register}
        onChange={onChange}
      />
      {maxLength && (
        <div className={styles.counter}>
          {watch()[name]?.length ?? 0} / {maxLength}
        </div>
      )}
    </div>
  );
};

export default TextArea;
