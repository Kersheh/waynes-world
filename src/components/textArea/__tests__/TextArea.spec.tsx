import React from 'react';
import { render } from '@testing-library/react';

import TextArea from '../TextArea';

jest.mock('react-hook-form', () => ({
  useFormContext: () => ({
    watch: jest.fn(),
    register: jest.fn()
  })
}));

describe('<TextArea/> component', () => {
  it('should render', () => {
    const { container } = render(<TextArea name="comments" />);

    expect(container).toMatchSnapshot();
  });
});
