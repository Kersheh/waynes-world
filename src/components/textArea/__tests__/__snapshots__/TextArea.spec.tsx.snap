// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArea/> component should render 1`] = `
<div>
  <div
    class="textArea"
  >
    <textarea
      name="comments"
    />
  </div>
</div>
`;
